tests/counters.rs

```diff
@@ -0,0 +1,232 @@
+// These tests cannot be run in parallel. We use the "serial_test" crate to run
+// them one by one.
+
+#[cfg(test)]
+mod counters {
+    use http::StatusCode;
+    use infinispan::Infinispan;
+    use reqwest::Response;
+    use serial_test::serial;
+    use std::collections::HashSet;
+    use std::iter::FromIterator;
+
+    #[tokio::test]
+    #[serial]
+    async fn create_weak_counter() {
+        cleanup().await;
+
+        let client = infinispan_client();
+        let counter_name = "test_counter";
+
+        client
+            .run(infinispan::request::counters::create_weak(counter_name))
+            .await
+            .unwrap();
+
+        let resp = client
+            .run(infinispan::request::counters::get(counter_name))
+            .await
+            .unwrap();
+
+        assert!(resp.status().is_success());
+        assert_eq!("0", resp.bytes().await.unwrap()); // Default counter value is 0
+    }
+
+    #[tokio::test]
+    #[serial]
+    async fn create_weak_counter_with_value() {
+        cleanup().await;
+
+        let client = infinispan_client();
+        let counter_name = "test_counter";
+        let counter_val = 10;
+
+        client
+            .run(infinispan::request::counters::create_weak(counter_name).with_value(counter_val))
+            .await
+            .unwrap();
+
+        let resp = client
+            .run(infinispan::request::counters::get(counter_name))
+            .await
+            .unwrap();
```

Comment:
This would look way better extracted, providing the request to the method and getting back the response. You could even have that helper also receive the expected response string and asserting the expectations.

Reply:
You mean extracting into a `run(request: Request)` function that also instantiates `client` ?

Reply:
Something like this?
```rust
pub async fn run(request: impl Into<Request>) -> Response {
    infinispan_client().run(request).await.unwrap()
}
```

I'd leave the part of checking the response for now, until we have a clearer idea of the helpers we want to provide to parse responses. I think this will become easier once we implement more parts of the API.

Reply:
Changed :+1: